src/Stratis.Bitcoin/Utilities/IpHelper.cs

```diff
@@ -7,32 +7,43 @@ namespace Stratis.Bitcoin.Utilities
 {
     public static class IpHelper
     {
+        // Don't re-use ports.
+        private static HashSet<int> usedPorts = new HashSet<int>();
+
+        public static int FindPort()
```

Comment:
@quantumagi to me, this is the only change that needs to be made. However, why don't we just store a set of used ports per `NodeBuilder` instance? And then check that collection?

So when you create a node builder instance:

```
            using (NodeBuilder builder = NodeBuilder.Create(this))
            {
```

the builder instance can have its own set of used ports? So return the port from `FindPort` and check if its already in use. If so, get another etc.

The set only needs to exist per node builder instance.

Reply:
We no longer need this new hashset as long as we perform retries when connections fail due to in-use ports. Also, keep in mind that even with you suggestion we can't avoid interference across test projects. See my other comments.